zerver/tests/test_event_system.py

```diff
@@ -828,6 +828,21 @@ def test_pronouns_field_type_support(self) -> None:
         [pronouns_field] = (field for field in custom_profile_fields if field["name"] == "Pronouns")
         self.assertEqual(pronouns_field["type"], CustomProfileField.PRONOUNS)
 
+    def test_unreads_case_insensitive_topics(self) -> None:
+        sender = self.example_user("hamlet")
+        self.login_user(sender)
+        self.send_stream_message(sender, "Denmark", "**hello**", topic_name="CASE DOES NOT MATTER")
+        self.send_stream_message(sender, "Denmark", "**bye**", topic_name="case does not matter")
```

Comment:
Do we have a test already that verifies this is case-*preserving*? That is, that the topic name emitted is one that some message was sent with, rather than something normalized like to lower case.

I see the implementation already looks correct on that front — it'd just be good to have a test to avoid someone accidentally breaking that in the future.

Reply:
Probably just changing this test to use mixed case in both examples (E.g. "CaSe/cAsE" for the first word) and add a couple asserts would suffice for exercising that.

Reply:
Repushed with the test so adjusted.